### PR TITLE
Merging the customImportMap and maps layer plugin into 1 single plugin.

### PR DESCRIPTION
**Pull request by navneet1v**
Merging the customImportMap and maps layer plugin into 1 single plugin.
Signed-off-by: Navneet Verma <navneev@amazon.com>

### Description
Merging the customImportMap and maps layer plugin into 1 single plugin.

Given that both the plugins are part of maps enhancements we are doing on OSD. It make sense to  merge both the plugins.

Currently we have done the file merge, but we will working on the repo to abstract the files in right folders. This is the first step towards merging the plugins. Improvement GH issue: https://github.com/opensearch-project/dashboards-maps/issues/167

This merge has the code till this commit: https://github.com/opensearch-project/dashboards-maps/commit/6dfdcbb347663ccc5b0828ea577a7f2ec6790958

### Next Steps:
1. Fixing the cypress tests which are failing due to wrong path.

### Issues Resolved
https://github.com/opensearch-project/dashboards-maps/issues/95

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: custom_import_map/public/plugin.tsx [hunk 1]
```diff
@@ -20,6 +28,39 @@ export class CustomImportMapPlugin
     core: CoreSetup,
     { regionMap }: AppPluginSetupDependencies
   ): CustomImportMapPluginSetup {
+    // Register an application into the side navigation menu
+    core.application.register({
+      id: PLUGIN_ID,
```
Comment (junqiu-lei) [inline custom_import_map/public/plugin.tsx:33]: Here the id will determine URL path of Maps,
Example: http://localhost:5603/fol/app/customImportMap/create

We are using `maps-dashboards`, How about change id to `maps` or `maps-dashboards`?

File: custom_import_map/public/plugin.tsx [hunk 2]
```diff
@@ -3,14 +3,22 @@
...
+  Plugin,
+} from '../../../src/core/public';
 import {
-  CustomImportMapPluginSetup,
-  CustomImportMapPluginStart,
-  AppPluginSetupDependencies,
+  AppPluginStartDependencies,
+  MapServices, CustomImportMapPluginSetup, CustomImportMapPluginStart,
 } from './types';
+import { PLUGIN_NAME, PLUGIN_ID, PLUGIN_NAVIGATION_BAR_TILE } from '../common';
```
Comment (junqiu-lei) [inline custom_import_map/public/plugin.tsx:19]: I think this is import from '../common/constants/shared'

**Review by junqiu-lei — DISMISSED** [review]

File: custom_import_map/public/plugin.tsx (hunk 1 above)
Comment (navneet1v) [reply custom_import_map/public/plugin.tsx]: fixed.

File: custom_import_map/public/plugin.tsx (hunk 2 above)
Comment (navneet1v) [reply custom_import_map/public/plugin.tsx]: fixed this and I have added the export statement in the index.ts file in that folder. So we can use both ../common and ../common/constants/shared

**Review by junqiu-lei — APPROVED** [review]

**Review by VijayanB — APPROVED** [review]

**Pull request merged**